lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1144,11 +1169,47 @@ instance DecodeAddress t => FromJSON (PostTransactionData t) where
 instance EncodeAddress t => ToJSON (PostTransactionData t) where
     toJSON = genericToJSON defaultRecordTypeOptions
 
+instance FromJSON PostWithdrawalData where
+    parseJSON = genericParseJSON defaultRecordTypeOptions
+instance ToJSON PostWithdrawalData where
+    toJSON = genericToJSON defaultRecordTypeOptions
+
+instance EncodeAddress t => ToJSON (PostPaymentOrWithdrawalData t) where
+    toJSON (PostPaymentOrWithdrawalData (Left x))  = toJSON x
+    toJSON (PostPaymentOrWithdrawalData (Right x)) = toJSON x
+instance DecodeAddress t => FromJSON (PostPaymentOrWithdrawalData t) where
+    parseJSON val = flip (withObject "PostPaymentOrWithdrawalData") val $ \o -> do
+        case (HM.lookup "source" o, HM.lookup "payments" o) of
+            (Just{}, Just{})   -> empty
+            (Nothing, Nothing) -> empty
+            (Just{}, Nothing)  ->
+                (PostPaymentOrWithdrawalData . Left)  <$> parseJSON val
+            (Nothing, Just{})  ->
+                (PostPaymentOrWithdrawalData . Right) <$> parseJSON val
+
 instance DecodeAddress t => FromJSON (PostTransactionFeeData t) where
     parseJSON = genericParseJSON defaultRecordTypeOptions
 instance EncodeAddress t => ToJSON (PostTransactionFeeData t) where
     toJSON = genericToJSON defaultRecordTypeOptions
 
+instance FromJSON PostWithdrawalFeeData where
+    parseJSON = genericParseJSON defaultRecordTypeOptions
+instance ToJSON PostWithdrawalFeeData where
+    toJSON = genericToJSON defaultRecordTypeOptions
+
+instance EncodeAddress t => ToJSON (PostPaymentOrWithdrawalFeeData t) where
+    toJSON (PostPaymentOrWithdrawalFeeData (Left x))  = toJSON x
+    toJSON (PostPaymentOrWithdrawalFeeData (Right x)) = toJSON x
+instance DecodeAddress t => FromJSON (PostPaymentOrWithdrawalFeeData t) where
+    parseJSON val = flip (withObject "PostPaymentOrWithdrawalFeeData") val $ \o -> do
```

Comment:
what about reusing what was done in impl of `instance DecodeAddress t => FromJSON (PostPaymentOrWithdrawalData t)` here?

Reply:
Because the JSON object isn't exactly the same (FeeData has no passphrase).